pkg/csi/service/fcd/controller.go

```diff
@@ -72,15 +75,133 @@ func (c *controller) CreateVolume(
 	req *csi.CreateVolumeRequest) (
 	*csi.CreateVolumeResponse, error) {
 
-	return nil, nil
+	secrets := req.GetSecrets()
+
+	// Volume Name
+	volName := req.GetName()
+
+	// Volume Size - Default is 10 GiB
+	volSizeBytes := int64(DefaultGbDiskSize * GbInBytes)
+	if req.GetCapacityRange() != nil && req.GetCapacityRange().RequiredBytes != 0 {
+		volSizeBytes = int64(req.GetCapacityRange().GetRequiredBytes())
+	}
+	volSizeMB := int64(volumeutil.RoundUpSize(volSizeBytes, GbInBytes)) * 1024
+
+	// Volume Type
+	datastoreType := vclib.TypeDatastoreCluster
+	volType := secrets[AttributeFirstClassDiskParentType]
+	if volType == string(vclib.TypeDatastore) {
+		datastoreType = vclib.TypeDatastore
+	}
+
+	datastoreName := secrets[AttributeFirstClassDiskParentName]
+
+	//check for required parameters
+	if secrets == nil {
+		return nil, status.Errorf(codes.Internal, "Create parameters is a required parameter.")
+	} else if len(volName) == 0 {
+		return nil, status.Errorf(codes.Internal, "Volume name is a required parameter.")
+	} else if len(secrets[AttributeFirstClassDiskParentType]) == 0 {
```

Comment:
Ah, yeah, this should be coming from the request `Parameters` field, not secrets.

Reply:
Switched to params for create